### ISSUE TITLE
Trailing comma in function signature after changing generic type to opaque type

### ISSUE THREAD

**Issue by fphilipe** [nicklockwood/SwiftFormat#1367]
SwiftFormat v0.50.8 fails to strip a comma when turning a generic type into an opaque type in a function signature that has another generic type that is left untouched:

```
cat <<EOF | swiftformat --fragment true --swiftversion 5.7
func f<T, U>(x: U) -> T where T: A, U: B {}
EOF
Running SwiftFormat...
func f<T>(x: some B) -> T where T: A, {}
SwiftFormat completed successfully.
```

Notice the comma before the opening brace:

```diff
-func f<T, U>(x: U) -> T where T: A, U: B {}
+func f<T>(x: some B) -> T where T: A, {}
```

This is not illegal, but still unexpected. Running SwiftFormat on the resulting output one more time does not result in a change, i.e. it's stable.

**Comment by nicklockwood** [comment]
Huh, I'm surprised that's legal - but yes, definitely not intentional. Thanks for reporting it.

**Comment by nicklockwood** [comment]
@fphilipe fixed in 0.50.9
